azure-pipelines.yml

```diff
@@ -18,35 +18,95 @@ pr:
 
 strategy:
   matrix:
-    linux-ghc-881:
+    # ghc-lib v8.8.1 (vanilla)
+
+    # ghc-8.4.4
+    linux-ghc-881-844:
       image: "Ubuntu 16.04"
       ghc-flavor: "ghc-8.8.1"
-    windows-ghc-881:
+      resolver: "lts-12.26"
+    windows-ghc-881-844:
       image: "vs2017-win2016"
       ghc-flavor: "ghc-8.8.1"
-    mac-ghc-881:
+      resolver: "lts-12.26"
+    mac-ghc-881-844:
       image: "macOS-10.13"
       ghc-flavor: "ghc-8.8.1"
+      resolver: "lts-12.26"
+
+    # ghc-8.6.5
+    linux-ghc-881-865:
+      image: "Ubuntu 16.04"
+      ghc-flavor: "ghc-8.8.1"
+      resolver: "lts-14.3"
+    windows-ghc-881-865:
+      image: "vs2017-win2016"
+      ghc-flavor: "ghc-8.8.1"
+      resolver: "lts-14.3"
+    mac-ghc-881-865:
+      image: "macOS-10.13"
+      ghc-flavor: "ghc-8.8.1"
+      resolver: "lts-14.3"
+
+    # ghc-lib v8.8.1 (DA)
 
-    linux-da-ghc-881:
+    # #ghc-8.4.4 (not useful - skipped)
+    # linux-da-ghc-881-844:
+    #   image: "Ubuntu 16.04"
+    #   ghc-flavor: "da-ghc-8.8.1"
+    #   resolver: "lts-12.26"
+    # windows-da-ghc-881-844:
+    #   image: "vs2017-win2016"
+    #   ghc-flavor: "da-ghc-8.8.1"
+    #   resolver: "lts-12.26"
+    # mac-da-ghc-881-844:
+    #   image: "macOS-10.13"
+    #   ghc-flavor: "da-ghc-8.8.1"
+    #   resolver: "lts-12.26"
+
+    # ghc-8.6.5
+    linux-da-ghc-881-865:
       image: "Ubuntu 16.04"
       ghc-flavor: "da-ghc-8.8.1"
-    windows-da-ghc-881:
+      resolver: "lts-14.3"
+    windows-da-ghc-881-865:
       image: "vs2017-win2016"
       ghc-flavor: "da-ghc-8.8.1"
-    mac-da-ghc-881:
+      resolver: "lts-14.3"
+    mac-da-ghc-881-865:
       image: "macOS-10.13"
       ghc-flavor: "da-ghc-8.8.1"
+      resolver: "lts-14.3"
+
+    # ghc-lib master (vanilla)
+
+    # ghc-8.4.4
+    linux-ghc-master-844:
+      image: "Ubuntu 16.04"
+      ghc-flavor: "ghc-master"
+      resolver: "lts-12.26"
```

Comment:
I am not entirely sold on just specifying the resolver here. That breaks quickly once you need some kind of extra-deps. Maybe we should just add a `stack-8.4.4.yaml` file to this repo and then instead of passing `--resolver` we can pass `--stack-yaml` which gives us more flexibility?

Reply:
I hear you but my understanding is, if a resolver is given it *overrides* the resolver in 'stack.yaml' and what's in the 'stack.yaml' otherwise applies including 'extra-deps' so I'm not sure that there is much extra flexibility to be had by doing that. On the other hand, if it proves useful in the future, the code is now easily adaptable to this approach so how about we take a wait and see?

Reply:
Not quite sure on the behavior of `resolver` but the extra flexibility becomes useful if you actually need different extra-deps per resolver which can happen quite easily.

Reply:
Ok. That's the bit that I wasn't sure in fact came up (extra deps on a per config basis). That being the case, agreed. Let's be lazy though and do it down the road when we have to :)